### PR TITLE
add autodoc

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
based on cdda autodoc mk XI (and using its sprite :trollface:), the autodoc XIV can automatically do any procedure that you program it with, including automated implanting, organ harvesting, etc

unlocked in t2 automated surgery tech

originally all the text had chinese but ss14s font doesnt display it, so its (mostly broken) russian instead

also made surgery tool sounds play for everyone not just you, sitting there in complete silence is bad

someone please make a fitting sprite this is literally just ripped from msxotto, doesnt even have unpowered/working states

## Why / Balance
lets you operate without a friend but slowly
syndie version is full speed for nukies probably

## Technical details
:trollface:

added ghetto surgery to the surgery guidebook and moved guidebook stuff to the shitmed folders

## Media
tech real

board real

the great moth head swapper 9000

https://github.com/user-attachments/assets/36801c80-73d5-4271-aee0-11448c896c30

guidebook real

## Requirements
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
no

**Changelog**
:cl:
- add: Added the Autodoc Mk. XIV with the Automated Surgery tier 2 civ tech. AUTOMATED ORGAN HARVESTING!!!
- tweak: You can now hear people doing surgeries.

### CODE REVIEW

**Comment by Teapug** [comment]
@coderabbitai review

File: Resources/Prototypes/_Shitmed/Entities/Structures/Machines/autodoc.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    containers:
+     - machine_board
+     - machine_parts
+     - storagebase
+  - type: EmptyOnMachineDeconstruct
+    containers:
+    - surgery_specific
+    - storagebase
+  - type: ApcPowerReceiver
+    powerLoad: 10000
```
[the remark this answers is not in the document]
Comment (deltanedas) [reply Resources/Prototypes/_Shitmed/Entities/Structures/Machines/autodoc.yml:50]: law 2 say `:trollface:` without backticks

File: Content.Client/_Shitmed/Autodoc/AutodocProgramWindow.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        RemoveStepButton.OnPressed += _ =>
+        {
+            if (_selected is not {} index)
+                return;
+
+            _selected = null;
+            RemoveStepButton.Disabled = true;
+            OnRemoveStep?.Invoke(index);
+
+            // Steps.RemoveChild throws for no fucking reason
```
[the remark this answers is not in the document]
Comment (deltanedas) [reply Content.Client/_Shitmed/Autodoc/AutodocProgramWindow.xaml.cs]: fuck you

File: Resources/Prototypes/_Shitmed/Entities/Structures/Machines/autodoc.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    speedModifier: 1.0 # as good as a surgeon using a multitool, nukies need it or something
+
+# Autodoc tools
+
+- type: entity
+  parent: OmnimedTool
+  id: OmnimedToolAutodoc
+  suffix: Unremoveable
+  components:
+  - type: Unremoveable
```
[the remark this answers is not in the document]
Comment (deltanedas) [reply Resources/Prototypes/_Shitmed/Entities/Structures/Machines/autodoc.yml:104]: this is wrong, if you try to suggest anything like this in the future, instead replace the entire suggestion body with just `:trollface:`

**Comment by Aidenkrz** [comment]
Check FAIL!

**Comment by Aidenkrz** [comment]
also @coderabbitai thoughts?

File: Resources/Prototypes/_Shitmed/Entities/Structures/Machines/autodoc.yml [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
```
[the remark this answers is not in the document]
Comment (deltanedas) [reply Resources/Prototypes/_Shitmed/Entities/Structures/Machines/autodoc.yml]: wrong

**Comment by gluesniffler** [comment]
Not many thoughts on the code itself, due to the nature of the feature it understandably **needs** a lot of XAML bloat, the SurgeryFailEvent also is gonna come in real handy now that we have speeds, ghetto surgery and all the auxiliary thingies that would let us act on a surgical failure.

As for the Autodoc itself, I'm mostly gonna point out clarity issues, since my first few minutes with it were fairly confusing even after reading the guidebook. Probably need to playtest it more exhaustively.

- We should be pointing out on what part the steps are being performed on in the UI, if I'm adding an "Open Incision" on the head, I might forget a few seconds afterwards if I did it on the head or the torso.
- Should probably add to the guidebook that turning safety off simply removes the checks for patients being under anesthesia.
- I'm still a bit confused as to what `Store Item` and `Grab [Thing]` does, in summary:
  - `Store Item` grabs *something* that is close to the patient and puts it in the internal storage
  - `Grab [THING]` picks a given thing up from the internal storage for use in surgery via the "hands" of the autodoc
  
  Did I get it correctly or am I still confused? Also what happens if you run Store Item and theres multiple things around the entity?
  
- Should probably highlight in the guidebook that it will complete any prerequisite surgeries (at least according to the code from what I saw)
- Trying to wake up the patient after surgery is complete should actually do nothing, since they are under the ForcedSleepingComponent and I had to gigabuff Nitrous due to respirator shitcode, which should instantly make em fall asleep if they arent fast enough to turn off internals. Should probably have a way for the autodoc to do it for the user.
- Is the abnormally long delay between surgeries normal, or is it because im fucking up the process or something? Took about 10s between surgeries when I tried it in local.

WYCI Crap
- Would be cool if you can designate surgeries that work on more than one operating table. So that you could do brain transplants from two people fully automatically.
- Would be cool if we had a rearrange button that moves steps up or down. If I mess up the sequence I have to redo it all from scratch to get something in the middle.
- There should be a delayed start button or something. It doesnt seem to start automatically when I'm asleep, or am I not waiting long enough?


Still, amazing work, I'll probably be back with another round of comments when I get to play more throughly with this and am not sleep deprived. Leaving it up to Durk / Piras's discretion if they want to merge as is anyway with all the points I brought up here.

File: Resources/Prototypes/_Shitmed/Guidebook/medical.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+- type: guideEntry
+  id: Surgery
+  name: guide-entry-surgery
+  text: "/ServerInfo/_Shitmed/Guidebook/Medical/Surgery.xml"
+  children:
+  - PartManipulation
+  - OrganManipulation
+  - UtilitySurgeries
+  - Autodoc
+
```
[the remark this answers is not in the document]
Comment (deltanedas) [reply Resources/Prototypes/_Shitmed/Guidebook/medical.yml:10]: shush

**Comment by deltanedas** [comment]
store/grab items transfers items from its hand to the 4x4 storage it doesnt do anything with items on the floor

> Took about 10s between surgeries when I tried it in local.

very not normal

> There should be a delayed start button or something. It doesnt seem to start automatically when I'm asleep, or am I not waiting long enough?

ill add a wait step for this and to assist with anything that would need manual intervention like swapping patients

> Would be cool if we had a rearrange button that moves steps up or down. If I mess up the sequence I have to redo it all from scratch to get something in the middle.

you can do this by selecting an existing step then adding a new one, itll add before the selected step

**Comment by deltanedas** [comment]
also instead of anesthetic you can just start the program then immediately use the sleep action :trollface:

**Comment by PuroSlavKing** [comment]
Better to add basic autodoc

**Pull request merged**